### PR TITLE
feat: Search

### PR DESCRIPTION
**Pull request by pklaschka**
Adds search to the `HTMLTheme`

### CODE REVIEW

File: src/processors/type.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { TypeAliasDeclaration } from 'ts-morph';
 import { processJsDocs } from './helpers/processJsDocs';
+import { ModuleTreeNode } from '../model';
 
 /**
  * Converts a `TypeAleasDeclaration` to a documentation-ready representation.
```
Comment (fussel178) [inline src/processors/type.ts:6]:
```suggestion
 * Converts a `TypeAliasDeclaration` to a documentation-ready representation.
```

File: src/processors/identifier.ts [hunk 2]
```diff
@@ -11,10 +12,12 @@ import { processNode } from './init';
  * processIdentifier(node);
  * ```
  */
-export function processIdentifier(node: Identifier): Record<string, unknown> {
+export function processIdentifier(
+	node: Identifier
+): ModuleTreeNode<ModuleTreeNode<unknown>> {
```
Comment (fussel178) [inline src/processors/identifier.ts:17]:
```suggestion
): ModuleTreeNode<ModuleTreeNode> {
```

File: src/processors/init.ts [hunk 3]
```diff
@@ -27,7 +28,7 @@ import { processInterfaceDeclaration } from './interface';
  * processNode(node);
  * ```
  */
-export function processNode(node: Node): Record<string, unknown> {
+export function processNode(node: Node): ModuleTreeNode<unknown> {
```
Comment (fussel178) [inline src/processors/init.ts:31]:
```suggestion
export function processNode(node: Node): ModuleTreeNode {
```

File: src/processors/module.ts [hunk 4]
```diff
@@ -14,22 +15,19 @@ import { processJsDocs } from './helpers/processJsDocs';
  */
 export function processModule(
 	node: NamespaceDeclaration
-): Record<string, unknown> {
+): ModuleTreeNode<ModuleTreeNode<unknown>> {
```
Comment (fussel178) [inline src/processors/module.ts:18]:
```suggestion
): ModuleTreeNode<ModuleTreeNode> {
```

File: src/processors/module.ts [hunk 5]
```diff
@@ -14,22 +15,19 @@ import { processJsDocs } from './helpers/processJsDocs';
...
 	if (node.hasNamespaceKeyword()) {
-		const res: Record<string, unknown> & { exportedMembers: unknown[] } = {
+		const res: ModuleTreeNode<ModuleTreeNode<unknown>> = {
```
Comment (fussel178) [inline src/processors/module.ts:20]:
```suggestion
		const res: ModuleTreeNode<ModuleTreeNode> = {
```

File: src/processors/variable.ts [hunk 6]
```diff
@@ -13,7 +14,7 @@ import { processJsDocs } from './helpers/processJsDocs';
  */
 export function processVariableDeclaration(
 	node: VariableDeclaration
-): Record<string, unknown> {
+): ModuleTreeNode<unknown> {
```
Comment (fussel178) [inline src/processors/variable.ts:17]:
```suggestion
): ModuleTreeNode {
```

**Review by fussel178 — CHANGES_REQUESTED** [review]
Some little TS type suggestions and a typing error.
Everything else LGTM :smile:

**Comment by pklaschka** [comment]
@fussel178 Thank you very much for your review. Yep, I changed it to default to `unknown` later in the process and forgot to update the references 👀 .

Kindly have a look again, it should now have been fixed, as per your suggestions 😉

**Review by fussel178 — APPROVED** [review]

**Pull request merged**